ui/server/src/main/scala/pl/touk/nussknacker/ui/process/deployment/DeploymentCommentValidator.scala

```diff
@@ -0,0 +1,57 @@
+package pl.touk.nussknacker.ui.process.deployment
+
+import cats.data.Validated
+import cats.data.Validated.{Invalid, Valid}
+import pl.touk.nussknacker.ui.listener.DeploymentComment
+
+object DeploymentCommentValidator {
+
+  def createDeploymentComment(comment: Option[String], settings: Option[DeploymentCommentSettings]): Validated[CommentValidationError, Option[DeploymentComment]] = {
+
+    (comment.filterNot(_.isEmpty), settings) match {
+      case (None, Some(_)) =>
+        Invalid(CommentValidationError("Comment is required."))
+      case (None, None) =>
+        Valid(None)
+      case (Some(comment), Some(deploymentCommentSettings)) =>
+        Validated.cond(
+          comment.matches(deploymentCommentSettings.validationPattern),
+          Some(DeploymentComment(comment)),
+          CommentValidationError(comment, deploymentCommentSettings))
+      case (Some(comment), None) =>
+        Valid(Some(DeploymentComment(comment)))
+    }
+  }
+
+  def unsafe(comment: String): DeploymentComment = DeploymentComment(comment)
+
+}
+
+case class CommentValidationError(message: String) extends Exception(message)
+
+object CommentValidationError {
+  def apply(comment: String, deploymentCommentSettings: DeploymentCommentSettings) = {
+    deploymentCommentSettings.exampleComment match {
+      case Some(exampleComment) =>
+        new CommentValidationError(s"Bad comment format '$comment'. Example comment: $exampleComment.")
+      case None =>
+        new CommentValidationError(s"Bad comment format '$comment'.")
```

Comment:
what do you mean? i thought the purpose of returning exampleComment was so that user does not receive regexp as his response for unsuccessful validation

Reply:
Yes but when smb not configure example comment IMO would be fine to at least show pattern. WDYT?

Reply:
ok, if it's for purpose of showing something to user when exampleComment is not specified I understand, added it